mlflow/pipelines/steps/register.py

```diff
@@ -26,18 +26,21 @@
 class RegisterStep(BaseStep):
     def __init__(self, step_config: Dict[str, Any], pipeline_root: str):
         super().__init__(step_config, pipeline_root)
-        self.tracking_config = TrackingConfig.from_dict(step_config)
+        self.tracking_config = TrackingConfig.from_dict(self.step_config)
```

Comment:
Is there a reason we need to force it to do self.step_config over just `step_config`?
I am just curious. Both ways should work, but I was curious if there some specific reasoning behind it.

Reply:
Theoretically, if `super()` mutates the config, it makes sense for us to capture those mutations when applying the tracking config. Seems reasonable to me.